lib/ansible/modules/user.py

```diff
@@ -3235,14 +3284,56 @@ def modify_user(self):
                             self.module.fail_json(name=self.name, msg=err, rc=rc)
 
         # Manage password
-        if self.update_password == 'always' and self.password is not None and info[1] != self.password:
-            cmd = [self.module.get_bin_path('chpasswd', True)]
-            cmd.append('--encrypted')
-            data = '{name}:{password}'.format(name=self.name, password=self.password)
-            rc, out, err = self.execute_command(cmd, data=data)
+        current_password = to_native(user_info[1])
+        new_password = self._build_password_string(current_password)
+        if self.update_password == 'always':
+            lock_status_mismatch = self.password_lock and not current_password.startswith('!')
+            password_changed = new_password != current_password
+            if lock_status_mismatch or password_changed:
+                cmd = [self.module.get_bin_path('chpasswd', True)]
+                cmd.append('--encrypted')
+                data = f'{self.name}:{new_password}'
+                rc, out, err = self.execute_command(cmd, data=data)
 
-            if rc is not None and rc != 0:
-                self.module.fail_json(name=self.name, msg=err, rc=rc)
+                if rc is not None and rc != 0:
+                    self.module.fail_json(name=self.name, msg=err, rc=rc)
+
+        # Manage user settings
+        uid = user_info[2]
+        if self.uid is not None:
+            uid = self.uid
+
+        passwd_entry = [
+            self.name,
+            'x',
+            to_native(uid),
+            to_native(gid),
+            self.comment or user_info[4],
+            self.home or user_info[5],
+            self.shell or user_info[6],
```

Comment:
Review from phone, so sorry if I missed it, but I can't see if you validate the shell against /etc/shells to validate that the shell exists?

Reply:
`adduser` will let you add any value for the shell. `chsh` only warns about a non-existent value but still returns 0.

Reply:
Then I see no problem with it, it would be nice to give a warning like chsh though

Reply:
https://github.com/ansible/ansible/pull/86342